### ISSUE TITLE
From version 0.7.1 `<Sprite.Provider>` inject scripts and styles inside own place (astro build phase)

### ISSUE THREAD

**Issue by le0pard** [natemoo-re/astro-icon#83]
Here example:

```js
---
import './app.css'
---
<html>
  <head>
  </head>
  <body>
    <Sprite.Provider>
      <Sprite pack="mdi" name="close" />
    </Sprite.Provider>
  </body>
</html>
```

From version 0.7.0:

```html
<html>
  <head>
    <link href="/_astro/style.0dedd145.css" rel="stylesheet">
  </head>
  <body>
    <svg aria-hidden="true" astro-icon-spritesheet="" style="position:absolute;width:0;height:0;overflow:hidden">...</svg>
  </body>
</html>
```

From version 0.7.1:

```html
<html>
  <head>
  </head>
  <body>
    <link href="/_astro/style.0dedd145.css" rel="stylesheet">
    <svg aria-hidden="true" astro-icon-spritesheet="" style="position:absolute;width:0;height:0;overflow:hidden">...</svg>
  </body>
</html>
```

Styles and scripts always injected at `<Sprite.Provider>` place, which lead to broken https://hotwired.dev/ usage - on each page body contain same css and js files, which need to re-apply in browser again. Inject directly to `<head>` not allow syntax (this is invalid):

```html
<html>
  <head>
    <Sprite.Provider>
  </head>
  <body>
      <Sprite pack="mdi" name="close" />
    </Sprite.Provider>
  </body>
</html>
```

**Comment by le0pard** [comment]
Here project which you can clone and reproduce this - https://github.com/rwpod/rwpod.github.io - just update `astro-icon` and rebuild it

**Comment by artstorm** [comment]
I've the same problem in my projects that uses Astro Icon v0.8.0. The sites were working fine until the release of Astro v2.0.7. That release had [many changes](https://github.com/withastro/astro/releases/tag/astro%402.0.7) related to prevent css injection in the body. It looks like those changes broke something with Astro Icon, as from Astro v2.0.7+,  using Astro Icon under some circumstances, now causes the injection of CSS and JS in the body instead.

For now, I've frozen my projects at Astro v2.0.6, as that is the last version that works together with Astro Icon for my sites, as I haven't had time to investigate this deeper, more than that verifying that disabling Astro Icon, injects the CSS in the head section again. What I did observe is that the more nested slots together with Astro Icon v0.8.0 and Astro v2.0.7+, the more likely this seems to happen.

**Comment by myakura** [comment]
Hi. I got the same issue in my project with Astro v2.0.7+ and astro-icon.
I made a few test cases and it appears if there's a component inside the `<Sprite.Provider>`, the link element is inserted right before the slotted elements.

https://myakura.github.io/test--astro-v2-0-7-and-astro-icon-issue/

**Comment by le0pard** [comment]
Looks like issue resolved if update astro to version `2.0.16`

**Comment by myakura** [comment]
yeah. the issue in Astro https://github.com/withastro/astro/issues/6318 was fixed in https://github.com/withastro/astro/pull/6363
